File: src/main/java/framework/ftc/cobaltforge/DirectiveChain.java
package framework.ftc.cobaltforge;

/**
 * Support for chaining directives
 * Created by Dummyc0m on 4/12/16.
 */
public final class DirectiveChain {
    private final CobaltForge forge;
    private AbstractDirective curr;

    DirectiveChain(CobaltForge forge, AbstractDirective curr) {
        this.forge = forge;
        this.curr = curr;
    }

    public final DirectiveChain then(AbstractDirective directive) {
        forge.addHandlerStore(directive);
        curr.setNext(directive);
        curr = directive;
        return this;
    }
}

File: src/main/java/framework/ftc/cobaltforge/DirectiveChain.kt
package framework.ftc.cobaltforge

/**
 * Support for chaining directives
 * Created by Dummyc0m on 4/12/16.
 */
class DirectiveChain internal constructor(private val forge: CobaltForge, private var curr: AbstractDirective?) {

    fun then(directive: AbstractDirective): DirectiveChain {
        forge.addHandlerStore(directive)
        curr!!.next = directive
        curr = directive
        return this
    }
}
